program/tests/replenish.rs

```diff
@@ -0,0 +1,431 @@
+            &accounts.alice_token,
+            &accounts.alice.pubkey(),
+            &accounts.alice.pubkey(),
+        );
+        let transaction = Transaction::new_signed_with_payer(
+            &instructions,
+            Some(&context.payer.pubkey()),
+            &[&context.payer, &accounts.alice],
+            context.last_blockhash,
+        );
+
+        let e = context
+            .banks_client
+            .process_transaction(transaction)
+            .await
+            .unwrap_err();
+        check_error(e, SinglePoolError::WrongStakeStake);
+    }
+
+    // onramp is already inactive but it doesnt have lamports for delegation
+    if fund_onramp {
+        let lamports = get_minimum_pool_balance(
+            &mut context.banks_client,
+            &context.payer,
+            &context.last_blockhash,
+        )
+        .await;
+
+        transfer(
+            &mut context.banks_client,
+            &context.payer,
+            &context.last_blockhash,
+            &accounts.onramp_account,
+            lamports,
+        )
+        .await;
+    }
+
+    // replenish and advance
+    replenish(&mut context, &accounts.vote_account.pubkey()).await;
+    advance_epoch(&mut context).await;
+
+    // deposit works in all cases
+    let instructions = instruction::deposit(
+        &id(),
+        &accounts.pool,
+        &accounts.alice_stake.pubkey(),
+        &accounts.alice_token,
+        &accounts.alice.pubkey(),
+        &accounts.alice.pubkey(),
+    );
+    let transaction = Transaction::new_signed_with_payer(
+        &instructions,
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &accounts.alice],
+        context.last_blockhash,
+    );
+
+    context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap();
+
+    assert!(context
+        .banks_client
+        .get_account(accounts.alice_stake.pubkey())
+        .await
+        .expect("get_account")
+        .is_none());
+
+    // onramp is active now if we transfered to it
+    let clock = context.banks_client.get_sysvar::<Clock>().await.unwrap();
+    let (_, onramp_stake, _) =
+        get_stake_account(&mut context.banks_client, &accounts.onramp_account).await;
+
+    // we require a fully active pool for any onramp state change to reduce complexity
+    // the pool for a healthy validator is never unstaked, and a fresh pool you can wait an epoch
+    // NOTE we might relax this for DepositSol, in which case this test would change
+    if fund_onramp && !reactivate_pool {
+        let stake = onramp_stake.unwrap();
+        assert_eq!(stake.delegation.activation_epoch, clock.epoch - 1);
+        assert_eq!(stake.delegation.deactivation_epoch, u64::MAX);
+    } else {
+        assert_eq!(onramp_stake, None);
+    }
+}
+
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+enum OnRampState {
+    Initialized,
+    Activating,
+    Active,
+    Deactive,
+}
```

Comment:
Can you add a deactivating case here for what we talked about in the PR?

Reply:
9a2d60b it requires its own special setup because the only way to actually deactivate it is to bonsai a reference vote history for `DeactivateDelinquent`